### ISSUE TITLE
Grid Items being destroyed and remounted upon initialization

### ISSUE THREAD

**Issue by zekedroid** [STRML/react-grid-layout#179]
It turns out items in the Grid are mounted twice for some reason. Has terrible side effects on some components loaded in the Grid. The following example creates a simple component to be used as the children of the Grid, and console logs the mounting. On load, it shows up twice.

Code to illustrate child components being mounted twice ([jsFiddle](http://jsfiddle.net/zekedroid/pkorny59/4/)):

```
const Grid = ReactGridLayout.WidthProvider(ReactGridLayout);

class TestComp extends React.Component {
	componentWillMount() {
  	console.log('component will mount');
  }
  
  render() {
  
  	return <div key={this.props.key} style={{ height: '100px', width: '100px', background: 'gray', margin: '10px' }}><span>Some content</span></div>;
   } 
}

class Logo extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            layouts: [
                {
                    i: '0',
                    x: 0,
                    y: 0,
                    w: 5,
                    h: 25,
                }, {
                    i: '1',
                    x: 5,
                    y: 0,
                    w: 7,
                    h: 30,
                },
            ]
        };
    }

    generateDOM() {
        const widgets = this.getWidgets();

        return widgets.map((widget, i) => {
            return (<div key={i} style={{ overflowY: 'auto' }}>
                <div style={{ background: 'gray' }} onClick={this.toggleWidget(widget.id).bind(this)}>
                    {widget.title}
                    {widget.component}
                </div>
            </div>);
        });
    }

    render() {
        const { layouts } = this.state;
        
        return (<div style={{ marginTop: '15px' }}>
                <Grid className="layout"
                  {...this.props}
                  layout={layouts}
                >
                    <TestComp key={'0'}/>
                    <TestComp key={'1'}/>
                </Grid>
            </div>);
    }
}

Logo.defaultProps = {
        isDraggable: true,
        isResizable: true,
        rowHeight: 1,
        cols: 12,
    }



React.render( <Logo /> , document.getElementById('container'));
```

**Comment by STRML** [comment]
The fiddle appears broken.

Try using rawgit.com to embed the build instead of just inlining it in the JS section.

Regardless, I am only seeing two mounts, which is expected because there are two components.

**Comment by zekedroid** [comment]
You're entirely right about the fiddle, it's just that our company blocked github for security reasons so I attempt these hacky solutions to importing libraries that jsfiddle doesn't let me import. Sorry about that.

Still, while attempting to recreate it in a better fashion, I fixed the fiddle and realized I couldn't recreate the bug. I looked back at my code and it's because I had left out one component, which I thought was irrelevant to the demo, that was actually the culprit. It's a component that "collapses" content smoothly using `react-motion`. This doesn't play well with the grid and I'm working to figure out why. It forces an unmount/mount when the layout changes. 

Probably not related to `react-grid-layout` so I'll close this. But for future reference, if anyone does see double mounting, check for `react-motion` or any other animation components in the items.
